SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -0,0 +1,806 @@
+    transport.hostName = self.ipAddress;
+    transport.portNumber = [NSString stringWithFormat:@"%d", self.tcpPort];
+    SDLProtocol *protocol = [[SDLProtocol alloc] init];
+    transport.delegate = protocol;
+    protocol.transport = transport;
+    [protocol.protocolDelegateTable addObject:self];
+
+    self.secondaryProtocol = protocol;
+    self.secondaryTransport = transport;
+
+    // we reuse Session ID acquired from primary transport's protocol
+    // this is for Register Secondary Transport frame
+    [self.secondaryProtocol storeHeader:self.primaryProtocolListener.primaryRPCHeader forServiceType:SDLServiceTypeControl];
+    // this is for video and audio services
+    [self.secondaryProtocol storeHeader:self.primaryProtocolListener.primaryRPCHeader forServiceType:SDLServiceTypeRPC];
+
+    [self.secondaryTransport connect];
+    return YES;
+}
+
+- (BOOL)sdl_startIAPSecondaryTransport {
+    SDLLogD(@"Starting Secondary Transport: iAP");
+
+    SDLIAPTransport *transport = [[SDLIAPTransport alloc] init];
+    SDLProtocol *protocol = [[SDLProtocol alloc] init];
+    transport.delegate = protocol;
+    protocol.transport = transport;
+    [protocol.protocolDelegateTable addObject:self];
+
+    self.secondaryProtocol = protocol;
+    self.secondaryTransport = transport;
+
+    // we reuse Session ID acquired from primary transport's protocol
+    // this is for Register Secondary Transport frame
+    [self.secondaryProtocol storeHeader:self.primaryProtocolListener.primaryRPCHeader forServiceType:SDLServiceTypeControl];
+    // this is for video and audio services
+    [self.secondaryProtocol storeHeader:self.primaryProtocolListener.primaryRPCHeader forServiceType:SDLServiceTypeRPC];
+
+    [self.secondaryTransport connect];
+    return YES;
+}
+
+- (BOOL)sdl_isTCPReady {
+    if ([self.ipAddress length] == 0) {
+        SDLLogD(@"IP address is empty");
+        return NO;
+    }
+    if (self.tcpPort < 0) {
+        SDLLogD(@"TCP port number is not available");
+        return NO;
+    }
+    if (!(self.tcpPort > 0 && self.tcpPort <= 65535)) {
+        SDLLogW(@"Invalid TCP port number for secondary transport: %d", self.tcpPort);
+        return NO;
+    }
+
+    if ([self sdl_getAppState] != UIApplicationStateActive) {
+        SDLLogD(@"App state is not Active, abort starting TCP transport");
+        return NO;
+    }
+
+    return YES;
+}
+
+- (UIApplicationState)sdl_getAppState {
+    if ([NSThread isMainThread]) {
+        return [UIApplication sharedApplication].applicationState;
+    } else {
+        __block UIApplicationState appState;
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            appState = [UIApplication sharedApplication].applicationState;
+        });
+        return appState;
+    }
+}
+
+
+#pragma mark - SDLProtocolListener Implementation
+
+// called on transport's thread, notifying that the transport is established
+- (void)onProtocolOpened {
+    SDLLogD(@"secondary transport connected");
```

Comment:
Capitalization

Reply:
Fixed in commit c6d6d98423e3372a89ad1f52331bd5f1b8c65fd6.